### PR TITLE
hexapdf: init at 1.0.2

### PR DESCRIPTION
**Pull request by bbenno**
## Description of changes

According to its [website](https://hexapdf.gettalong.org/), hexapdf is a _versatile PDF creation and manipulation library for Ruby_. 
It can be used as a CLI tool for manipulating or optimizing PDF files.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/he/hexapdf/package.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  inherit ruby;
+  gemdir = ./.;
+
+  passthru.updateScript = bundlerUpdateScript "hexapdf";
+
+  meta = with lib; {
+    description = "Versatile PDF creation and manipulation library";
+    homepage = "https://hexapdf.gettalong.org/";
+    changelog = "https://github.com/gettalong/hexapdf/blob/master/CHANGELOG.md";
+    license = licenses.agpl3Only;
```
Comment (bbenno) [inline pkgs/by-name/he/hexapdf/package.nix:16]: I am uncertain, whether this is correct.
It is released under AGPL Open Source, a commercial licence available (see also https://gettalong.at/hexapdf/).
Do we need to include the commercial option here?

Related links:
- Website about commercial license: https://gettalong.at/hexapdf/
- Custom LICENSE file in source code repository: https://github.com/gettalong/hexapdf/blob/master/LICENSE
- official website: https://hexapdf.gettalong.org/

Comment (bbenno) [reply pkgs/by-name/he/hexapdf/package.nix]: The Gem contains [1](https://github.com/gettalong/hexapdf/blob/e784ea875a60e04512bcbb28fd99c79044ad9941/hexapdf.gemspec#L33)

```ruby
s.licenses = ['AGPL-3.0', 'Nonstandard']
```

so without overseeing possible consequences, I'd propose 

```nix
license = with licenses; [ agpl3Only unfree ];

Comment (FliegendeWurst) [reply pkgs/by-name/he/hexapdf/package.nix]: The website states:
> Everything that is available in the commercial version is also available in the AGPL version

So I would say agpl3Only is fine.

**Comment by bbenno** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 313917`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hexapdf</li>
  </ul>
</details>

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**